From 1b1f0201e835bc0312dbcea450ba47357ab08e40 Mon Sep 17 00:00:00 2001
From: Fakeman <55951275+fakemanoan@users.noreply.github.com>
Date: Mon, 25 Sep 2023 12:54:11 +0100
Subject: [PATCH 2/2] Add back TARGET_HAS_MEMFD_BACKPORT

We require this patch as in android it checks for kernel versions 3.18 or newer for memfd support. We are on 3.10 for the foreseeable future.

Add flag to skip kernel version check (we have it in the kernel and it didnt cause issues before so its probably fine)

Taken from cherishos-UL

Change-Id: I749e8a1b4e145adc9ff49feec109bf3dfb16d783
---
 build/soong/Android.bp     | 17 +++++++++++++++++
 config/BoardConfigSoong.mk |  2 ++
 2 files changed, 19 insertions(+)

diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index b8f9f6e2..c3a96549 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -238,6 +238,23 @@ needs_netd_direct_connect_rule {
     },
 }
 
+soong_config_module_type {
+    name: "has_memfd_backport",
+    module_type: "cc_defaults",
+    config_namespace: "cherishGlobalVars",
+    bool_variables: ["has_memfd_backport"],
+    properties: ["cppflags"],
+}
+
+has_memfd_backport {
+    name: "has_memfd_backport_defaults",
+    soong_config_variables: {
+        has_memfd_backport: {
+            cppflags: ["-DHAS_MEMFD_BACKPORT"],
+        },
+    },
+}
+
 soong_config_module_type {
     name: "bootloader_message_offset",
     module_type: "cc_defaults",
diff --git a/config/BoardConfigSoong.mk b/config/BoardConfigSoong.mk
index e98e7531..565170d5 100644
--- a/config/BoardConfigSoong.mk
+++ b/config/BoardConfigSoong.mk
@@ -32,6 +32,7 @@ SOONG_CONFIG_cherishGlobalVars += \
     additional_gralloc_10_usage_bits \
     bootloader_message_offset \
     needs_netd_direct_connect_rule \
+    has_memfd_backport \
     camera_override_format_from_reserved \
     gralloc_handle_has_custom_content_md_reserved_size \
     gralloc_handle_has_reserved_size \
@@ -100,6 +101,7 @@ SOONG_CONFIG_cherishGlobalVars_aapt_version_code := $(shell date -u +%Y%m%d)
 SOONG_CONFIG_cherishGlobalVars_additional_gralloc_10_usage_bits := $(TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS)
 SOONG_CONFIG_cherishGlobalVars_bootloader_message_offset := $(BOOTLOADER_MESSAGE_OFFSET)
 SOONG_CONFIG_cherishGlobalVars_needs_netd_direct_connect_rule := $(TARGET_NEEDS_NETD_DIRECT_CONNECT_RULE)
+SOONG_CONFIG_cherishGlobalVars_has_memfd_backport := $(TARGET_HAS_MEMFD_BACKPORT)
 SOONG_CONFIG_cherishGlobalVars_target_health_charging_control_charging_path := $(TARGET_HEALTH_CHARGING_CONTROL_CHARGING_PATH)
 SOONG_CONFIG_cherishGlobalVars_target_health_charging_control_charging_enabled := $(TARGET_HEALTH_CHARGING_CONTROL_CHARGING_ENABLED)
 SOONG_CONFIG_cherishGlobalVars_target_health_charging_control_charging_disabled := $(TARGET_HEALTH_CHARGING_CONTROL_CHARGING_DISABLED)
-- 
2.25.1

